admin/src/js/controllers/edit-user.js

```diff
@@ -409,6 +420,15 @@ angular
         });
     };
 
+    $scope.toggleRole = (role) => {
+      const index = $scope.editUserModel.roles.indexOf(role);
+      if (index === -1) {
+        $scope.editUserModel.roles.push(role);
+      } else {
+        $scope.editUserModel.roles.splice(index, 1);
+      }
```

Comment:
I don't want to mess with the model too much (sorting would cause angular to re-render, right?). We could sort before submitting, or in the API before storing, or in the API before returning... What's the concern here? That we don't want to update a user doc that hasn't actually changed in any meaningful way?

Reply:
I've rearranged the code slightly to do the sorting in place, and added a test that this works.